proctor-common/src/main/java/com/indeed/proctor/common/TestRangeSelector.java

```diff
@@ -156,6 +153,23 @@ private int findMatchingRuleInternal(
         return -1;
     }
 
+    private boolean isValidAllocation(final String identifier, final String rule) {
+        return testDefinition.getTestType().equals(TestType.RANDOM)
+                || isNormalAllocation(identifier)
+                || isUnitlessAllocation(rule);
+    }
+
+    private boolean isNormalAllocation(final String identifier) {
+        return identifier != null
+                && identifierValidator.validate(testDefinition.getTestType(), identifier);
+    }
+
+    private boolean isUnitlessAllocation(final String rule) {
+        return rule != null
```

Comment:
We can use StringUtils.hasText(rule) as well. But it's NIT as we have a check below .contains("missingExperimentalUnit")

Reply:
I think hasText is not needed as the contains will handle empty rules.